runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -961,6 +967,13 @@ Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv *env, jclass clazz, job
 
 					new_clazz = J9VM_J9CLASS_TO_HEAPCLASS(declaringClass);
 					new_flags = MN_IS_FIELD | (fieldID->field->modifiers & CFR_FIELD_ACCESS_MASK);
+					if (J9_ARE_ALL_BITS_SET(new_flags, J9AccFinal)) {
+						J9ROMClass *romClass = fieldID->declaringClass->romClass;
+						if (J9_ARE_ALL_BITS_SET(new_flags, J9AccStatic) || J9ROMCLASS_IS_HIDDEN(romClass) || J9ROMCLASS_IS_RECORD(romClass)) {
```

Comment:
Same question - and this might be a good place to add an abstraction layer to centralize this info.  A new `VMHelpers::isJLITrustedFinal()` might be the cleanest way to go

Reply:
will move this check into `VMHelpers`